lib/node/node.js

```diff
@@ -472,43 +470,10 @@ function validate(validator, params, expectedParamName) {
     return param;
 }
 
-Node.prototype.computeRequirements = function(databaseService, limits, callback) {
-    var default_limit = null; // no limit by default
-
-    // The default comptueRequirements methos computes the number estimated of output rows of the node.
-    // Two possible approaches to do this are:
-    // 1. use the node's query, e.g.
-    //        this.requirements.setEstimatedNumberOfRowsFromQuery(databaseService, default_limit, limits, callback);
-    //    (we could also use setNumberOfRowsFromQuery for the exact number of rows)
-    // 2. reduce overhead by basing the estimation on the previously computed estimations of the input nodes;
-    //    here we would use some simple, common heuristic, such as:
-    //        this.requirements.setMaxInputNumberOfRows(limits, default_limit);
-    //        return callback(null);
-    //    And we'd need to override this for node classes for which this not acceptable
-
-    this.requirements.setMaxInputNumberOfRows(limits, default_limit);
-    return callback(null);
-};
-
-Node.prototype.requirementMessages = function() {
-    var messages = [];
-
-    // Note that any falsy value (null, 0, '', ...) means no limits:
-    if (this.requirements.getLimit('maximumNumberOfRows')) {
-        if (this.requirements.getEstimatedRequirement('numberOfRows') >
-            this.requirements.getLimit('maximumNumberOfRows')) {
-            messages.push('too many result rows');
-        }
-    }
-
-    return messages;
-};
-
-Node.prototype.validateRequirements = function() {
-    var messages = this.requirementMessages();
-    var err = null;
-    if (messages.length > 0) {
-        err = new Error(messages.join('\n'));
-    }
-    return err;
+Node.prototype.checkLimits = function(databaseService, limits, logger, callback) {
```

Comment:
What about passing a collaborator that already is aware of `globalLimits`, `databaseService`, `logger`, so you don't have to care/understand that much about what the dependencies for each NodeRequirements method are.